pkg/agent/common/backoff/backoff.go

```diff
@@ -30,7 +35,23 @@ func NewBackoff(clk clock.Clock, interval time.Duration) BackOff {
 		MaxInterval:         _maxIntervalMultiple * interval,
 		MaxElapsedTime:      _noMaxElapsedTime,
 	}
+	for _, opt := range opts {
+		opt.applyOption(b)
+	}
 	b.Reset()
 
 	return b
 }
+
+// WithMaxInterval returns maxInterval backoff option to override the MaxInterval
+func WithMaxInterval(maxInterval time.Duration) Option {
+	return backoffOption{maxInterval: maxInterval}
+}
+
+type backoffOption struct {
+	maxInterval time.Duration
+}
+
+func (b backoffOption) applyOption(backoff *backoff.ExponentialBackOff) {
+	backoff.MaxInterval = b.maxInterval
+}
```

Comment:
```suggestion
func (b backoffOption) applyOption(bo *backoff.ExponentialBackOff) {
	bo.MaxInterval = b.maxInterval
}
```
Using the variable name `backoff` obscures the import of the `github.com/cenkalti/backoff/v3` package, please use a different variable name to disambiguate

Reply:
Updated